File: ai/book_note/Hello-Agents/code/HelloAgents/helloagents-trip-planner/frontend/src/services/trip.ts
import { api } from "./api";

import type { TripRequest, ValidatedTripRequest } from "../types/trip";

export async function validateTripRequest(
  request: TripRequest,
): Promise<ValidatedTripRequest> {
  const response = await api.post<ValidatedTripRequest>(
    "/trip/validate",
    request,
  );
  return response.data;
}

File: ai/book_note/Hello-Agents/code/HelloAgents/helloagents-trip-planner/backend/app/api/routes/trip.py
"""Travel request validation routes introduced in section 13.2."""

from __future__ import annotations

from fastapi import APIRouter

from ...models import TripRequest


router = APIRouter(prefix="/trip", tags=["trip"])


@router.post("/validate", response_model=TripRequest)
def validate_trip_request(request: TripRequest) -> TripRequest:
    """Return a normalized request without invoking agents or external APIs."""
    return request
